### PR TITLE
CRM457-2322 part 2: Push record to UAT

### PR DESCRIPTION
**Pull request by patrick-laa**
## Description of change
- Move anonymised records to UAT instead of localhost
- Allow deleting from UAT by rake task

[Link to relevant ticket](https://dsdmoj.atlassian.net/browse/CRM457-2322)

### CODE REVIEW

**Review by elken — CHANGES_REQUESTED** [review]
I don't think we should _replace_ the existing download tool, instead maybe handle optional arguments (and default to local, so we don't accidentally spam UAT with claims)

File: bin/delete_anonymised [hunk 1]
```diff
@@ -1,5 +1,12 @@
 #!/bin/sh
 
 LAA_REF=$1
+DESTINATION_ENVIRONMENT=$2
 
-bundle exec rails anonymised:delete\[$LAA_REF\]
+if [[ $DESTINATION_ENVIRONMENT == "local" ]]; then
+  bundle exec rails anonymised:delete\[$LAA_REF\]
```
Comment (elken) [inline bin/delete_anonymised:7]: So we can just do `if != local` here and put the `kubectl` command in that as both conditions run the same line

File: bin/download_anonymised [hunk 2]
```diff
@@ -2,13 +2,21 @@
...
-echo "Importing anonymised record..."
-ANONYMISED_DOWNLOAD_OUTPUT=$OUTPUT bundle exec rails anonymised:import\[ANONYMISED_DOWNLOAD_OUTPUT\]
+echo "Importing anonymised record to $2..."
+
+if [[ $DESTINATION_ENVIRONMENT == "local" ]]; then
+  ANONYMISED_DOWNLOAD_OUTPUT=$OUTPUT bundle exec rails anonymised:import\[ANONYMISED_DOWNLOAD_OUTPUT\]
+else
+  kubectl exec deploy/laa-crime-application-store-app \
+  -it -n laa-crime-application-store-$DESTINATION_ENVIRONMENT -- \
+  env ANONYMISED_DOWNLOAD_OUTPUT="${OUTPUT}" bundle exec rails anonymised:import\[ANONYMISED_DOWNLOAD_OUTPUT\]
```
Comment (elken) [inline bin/download_anonymised:19]: Same here, `env` isn't needed but the shell escape quoting should be

File: bin/delete_anonymised (hunk 1 above)
Comment (patrick-laa) [reply bin/delete_anonymised:7]: Ah but the kubectl command and the `bundle exec rails` aren't separate commands, they're the same one with a line break. I don't think you can if-block a partial statement

Comment (elken) [reply bin/delete_anonymised:7]: Oh I see the line breaks now, yeah fine

File: bin/download_anonymised (hunk 2 above)
Comment (patrick-laa) [reply bin/download_anonymised:19]: FWIW env really is needed - without it you get `error: Internal error occurred: error executing command in container: failed to exec in container: failed to start exec...`. You may not have spotted that line 19 is not a standalone statement, it's a continuation of the statement starting on line 17, and is therefore  run in the kube pod and needs slightly special escaping to get it there.

Comment (elken) [reply bin/download_anonymised:19]: Yeah line break again, it is. Line 15 should have the same shell escape quoting though

**Review by elken — APPROVED** [review]

File: bin/download_anonymised [hunk 3]
```diff
@@ -16,7 +16,7 @@ if [[ $DESTINATION_ENVIRONMENT == "local" ]]; then
 else
   kubectl exec deploy/laa-crime-application-store-app \
   -it -n laa-crime-application-store-$DESTINATION_ENVIRONMENT -- \
-  env ANONYMISED_DOWNLOAD_OUTPUT="${OUTPUT}" bundle exec rails anonymised:import\[ANONYMISED_DOWNLOAD_OUTPUT\]
+  env ANONYMISED_DOWNLOAD_OUTPUT="$OUTPUT" bundle exec rails anonymised:import\[ANONYMISED_DOWNLOAD_OUTPUT\]
```
Comment (elken) [inline bin/download_anonymised:19]: That's the opposite of what I meant :P Escaping it prevents something in the output causing it to be interpreted as part of the script, so you DO want them both to be `"${OUTPUT}"`

**Pull request merged**